docs/architecture/COSMWASM_POOLS.MD

```diff
@@ -54,6 +54,34 @@ One caveat on utilizing cw2 information is that there is no uniqueness check for
 `crates.io:transmuter` `<3.0.0`: requires no additional information.
 `crates.io:transmuter` `>=3.0.0`: requires alloyed asset denom and normalization factors for each asset.
 
+### Alloyed
```

Comment:
Note to reviewer:

Upon regenerating mainnet test state, another bug stemming from the treatment of alloyed denoms resurfaced.

It stemmed from the inconsistency in handling the alloyed LP share.

As an outcome, I defined the invariants and fixed the tests in [this PR](https://github.com/osmosis-labs/sqs/pull/377).

#377 was merged here to unblock the CI pipeline.

For ease of review, I recommend starting from #377 and then coming back to reviewing the remaining work here.